### PR TITLE
build: reset build to use TypeScript 2.6.x

### LINKED ISSUE #638: AOT Builds fail with Angular 5.2.7
Delete `package-lock.json` and run `npm i && npm run lib:build:aot`
See failures in terminal.

> Works fine in Angular 5.2.6.

### PR DESCRIPTION
**Pull request by CaerusKaru**
* Lock version of TypeScript to 2.6.x since Angular doesn't support
  v2.7 yet
* Re-add MediaQueriesModule as a deprecated alias for CoreModule
* Fix AOT/Universal tasks to check for presence of lock file before
  deleting it

Fixes #638

### CODE REVIEW

File: src/lib/flex/flex-offset/flex-offset.ts [hunk 1]
```diff
@@ -156,7 +162,7 @@ export class FlexOffsetDirective extends BaseFxDirective implements OnInit, OnCh
     this._applyStyleToElement(this._buildCSS(value));
   }
 
-  protected _buildCSS(offset): { [x: string]: string } {
+  protected _buildCSS(offset): StyleDefinition {
```
Comment (ThomasBurleson) [inline src/lib/flex/flex-offset/flex-offset.ts]: Good improvement.

File: src/apps/demo-app/package.json [hunk 2]
```diff
@@ -17,7 +17,7 @@
     "@angular/common": "file:../../../node_modules/@angular/common",
     "@angular/compiler": "file:../../../node_modules/@angular/compiler",
     "@angular/core": "file:../../../node_modules/@angular/core",
-    "@angular/flex-layout": "file:../../../dist/releases/flex-layout",
+    "@angular/flex-layout": "file:../../../dist/releases/flex-layout/angular-flex-layout.tgz",
```
Comment (ThomasBurleson) [inline src/apps/demo-app/package.json]: Why is a `.tgz` needed here ?

File: tools/gulp/tasks/aot.ts [hunk 3]
```diff
@@ -1,20 +1,46 @@
...
+  'aot:bundle:rename',
+  'aot:clean',
+  'aot:deps',
+  'aot:add:tar')
+);
 
 task('aot:deps', [], execTask(
   'npm', ['install'], {cwd: demoAppSource}));
 
+task('aot:add:tar', [], execTask(
```
Comment (ThomasBurleson) [inline tools/gulp/tasks/aot.ts]: Please add comment paragraph here to explain your intent.

File: tools/gulp/tasks/aot.ts [hunk 4]
```diff
@@ -29,12 +55,13 @@ task('aot:clear:mods', [], execTask(
   }
 ));
 
-task('aot:clear:lock', [], execTask(
+task('aot:clear:lock', [], existsSync(join(demoAppSource, 'package-lock.json')) ?
+  execTask(
```
Comment (ThomasBurleson) [inline tools/gulp/tasks/aot.ts]: Let's do this here also:

```ts
 () => existsSync(join(universalAppSource, 'package-lock.json')) ?
```

> Similar to the fix in Universal.

File: tools/gulp/tasks/universal.ts [hunk 5]
```diff
@@ -74,12 +75,15 @@ task('prerender:clear:deps', [], execTask(
   }
 ));
 
-task('prerender:clear:lock', [], execTask(
-  'rm', ['./package-lock.json'], {
-    silent: true,
-    cwd: universalAppSource
-  }
-));
+task('prerender:clear:lock', [], () => existsSync(join(universalAppSource, 'package-lock.json')) ?
```
Comment (ThomasBurleson) [inline tools/gulp/tasks/universal.ts]: Very nice!  
> I did not think of this approach.

File: src/apps/demo-app/package.json (hunk 2 above)
Comment (CaerusKaru) [reply src/apps/demo-app/package.json]: To maintain parity with the Universal task, you'll see it has the same logic. There was some reason in the Universal app why direct directory access wasn't working (I think a mismatch in Angular versioning/linking issue), so installing it was the only way to get it to function correctly.

File: tools/gulp/tasks/aot.ts (hunk 3 above)
Comment (CaerusKaru) [reply tools/gulp/tasks/aot.ts]: As cited above, but will add to the file

File: tools/gulp/tasks/aot.ts [hunk 6]
```diff
@@ -1,20 +1,53 @@
...
 
 task('aot:deps', [], execTask(
   'npm', ['install'], {cwd: demoAppSource}));
 
+
+/**
+ * The following tasks bundle Flex Layout into a tar file that can be installed
+ * directly instead of linked to via sym link. When linking, there are some issues
+ * that npm introduces, like Angular functionality impairment. This also has the
+ * benefit of better simulating the install process for Flex Layout in a CLI app
```
Comment (ThomasBurleson) [inline tools/gulp/tasks/aot.ts]: Nice.

**Comment by ThomasBurleson** [comment]
Merged with SHA 95a6e83

**Pull request closed without merging**